spec/requests/posts_spec.rb

```diff
@@ -1,7 +1,25 @@
 require 'rails_helper'
 
 RSpec.describe 'Posts', type: :request do
-  describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
+  describe 'GET /index/:user_id/posts' do
+    it 'renders the index template' do
+      user = User.create(name: 'Steven', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from jos.',
+                         posts_count: 0)
+      get user_posts_path(user_id: user.id)
+      expect(response).to have_http_status(:ok)
+      expect(response).to render_template(:index)
+      expect(response.body).to include('User Posts')
+    end
+  end
+  describe 'GET /users/:user_id/posts/:id' do
+    it 'renders the show template' do
+      user = User.create(name: 'steven', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from jos',
+                         posts_count: 0)
+      post = Post.create(title: 'post1', author: user, likes_count: 0, comments_count: 0, created_at: 5.day.ago)
+      get user_posts_path(user_id: user.id, id: post.id)
+      expect(response).to have_http_status(:ok)
+      expect(response).to render_template(:show)
+      expect(response.body).to include('User Posts')
+    end
   end
 end
```

Comment:
Please note that some of your tests are not passing, kindly fix them 👍🏽
![Screenshot from 2023-08-24 21-40-48](https://github.com/emmanuelsteven/MyBlogApp/assets/44624138/a1290025-155d-4851-87a0-1d61c1e22d68)